src/include/iscsi-lio-server/TargetData.rb

```diff
@@ -561,31 +558,47 @@ def fetch_mutual_password
     @discovery_auth.fetch("mutual_password")
   end
 
+  # Query discovery authentication data from the system and store the
+  # data in the object.
   def analyze
-    cmd = "targetcli iscsi/ get discovery_auth enable"
-    cmd_out = `#{cmd}`
-    status = cmd_out[7,cmd_out.length]
-    store_status(status)
-
-    cmd = "targetcli iscsi/ get discovery_auth userid"
-    cmd_out = `#{cmd}`
-    userid = cmd_out[7,cmd_out.length]
-    store_userid(userid)
-
-    cmd = "targetcli iscsi/ get discovery_auth password"
-    cmd_out = `#{cmd}`
-    password = cmd_out[9,cmd_out.length]
-    store_password(password)
-
-    cmd = "targetcli iscsi/ get discovery_auth mutual_userid"
-    cmd_out = `#{cmd}`
-    mutual_userid = cmd_out[14,cmd_out.length]
-    store_mutual_userid(mutual_userid)
-
-    cmd = "targetcli iscsi/ get discovery_auth mutual_password"
-    cmd_out = `#{cmd}`
-    mutual_password = cmd_out[16,cmd_out.length]
-    store_mutual_password(mutual_password)
+    restricted_recorder = RestrictedRecorder.new(Yast::Y2Logger.instance)
+    output = Yast::Execute.locally!("/usr/bin/targetcli", "iscsi/ get discovery_auth enable userid "\
+      "password mutual_userid mutual_password", stdout: :capture, recorder: restricted_recorder)
+
+    lines = output.split("\n")
+
+    analyze_lines(lines, "enable=") { |value| store_status(value.strip == "True") }
```

Comment:
Just some notes to make it more robust:

- What about case insensitive comparison? I'm afraid that they could change the output to `true` at some point and it would break... :worried: (`value.strip.casecmp("true") == 0`)
- What about possible translations? I guess this tool does not use any translated output, but maybe we should explicitly set the English (`en_US.UTF-8`) or the `C` locale to ensure it does not break in other languages.

Reply:
Added suggestions.